### PR TITLE
Eligibility Check - Filter by subject

### PR DESCRIPTION
**Pull request by steventux**
## Trello

https://trello.com/c/p5UVMi59/1432-build-new-subject-criteria-in-eligibility-checker

## Context

From 1 Feb, applicants from the some countries will only be able to apply/be awarded QTS if they are qualified to teach Maths, Languages ( French, German, Italian, Japanese, Latin, Mandarin, Russian, Spanish) and Sciences (physics, chemistry, Biology, general science)

## Changes

- Teach children question presents a different age range (slightly different content) if the teacher has selected a relevant country.
image

- A "Yes" answer takes them to a question about which subject they are qualified in.
image

- Answering "No" in either case takes them through the rest of the eligibility questions ending with the _Ineligible_ outcome and presents relevant reasons: 
image

### CODE REVIEW

File: app/views/eligibility_interface/qualified_for_subject/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      <li>Latin</li>
+      <li>Mandarin</li>
+      <li>Russian</li>
+      <li>Spanish</li>
+    </ul>
+  </div>
+</div>
+
+<%= form_with model: @qualified_for_subject_form, url: eligibility_interface_qualified_for_subject_url, method: :post do |f| %>
+  <%= f.govuk_error_summary %>
```
Comment (thomasleese) [inline app/views/eligibility_interface/qualified_for_subject/new.html.erb:35]: I think we should move this to the top of the file so any error messages appear right at the top.

File: app/views/eligibility_interface/qualified_for_subject/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    :qualified_for_subject,
+    [OpenStruct.new(label: 'Yes', value: true), OpenStruct.new(label: 'No', value: false)],
+    :value,
+    :label,
+    legend: nil,
+  ) %>
+  <%= f.govuk_submit prevent_double_click: false %>
+<% end %>
+
+<details class="govuk-details" data-module="govuk-details">
```
Comment (thomasleese) [inline app/views/eligibility_interface/qualified_for_subject/new.html.erb:47]: I'd suggest using the [`govuk_details`](https://govuk-components.netlify.app/components/details/) component here.

**Review by thomasleese — COMMENTED** [review]
Two small comments, other than that it all looks good to me.

**Review by thomasleese — APPROVED** [review]

**Comment by steventux** [comment]
@thomasleese I've amended this PR to continue through the EC flow in the case of No answers to the new filter by subject questions. The last 2 commits were affected.

**Pull request merged**